pkg/build/builder/common.go

```diff
@@ -82,3 +92,124 @@ func updateBuildRevision(c client.BuildInterface, build *api.Build, sourceInfo *
 		glog.Warningf("An error occurred saving build revision: %v", err)
 	}
 }
+
+// randomBuildTag generates a random tag used for building images in such a way
+// that the built image can be referred to unambiguously even in the face of
+// concurrent builds with the same name in the same namespace.
```

Comment:
A tag passed to `docker build` and a container name to `docker run` (or their Remote API equivalents) behave differently.

In the container name case, trying to create a container with the same name will fail with an error. On the other hand, a build with an existing tag will proceed normally and overwrite the tag.

Using a deterministic name for the build tag would kill the purpose of being able to identify with certainty what image has been produced by the current build.